redash/query_runner/google_analytics.py

```diff
@@ -164,7 +164,7 @@ def run_query(self, query, user):
             params = json_loads(query)
         except:
             params = parse_qs(urlparse(query).query, keep_blank_values=True)
-            for key in params.keys():
+            for key in [*params]:
```

Comment:
I'm not sure this is the issue:

```
>>> d = {'ab': 'yesterday', 'cd': 2}
>>> for k in d.keys():
...     print(k)
... 
ab
cd
```

(this shows that the keys are iterated properly)

*But*, check the following:

```
>>> ",".join("yesterday")
'y,e,s,t,e,r,d,a,y'
```

So the issue is actually line 168 which applies `",".join` on any value.

Reply:
I agree, but subsequently, using `[*params]` would resolve this.

Reply:
The real problem is line 170, which messes around with the `dict_keys` iterator, something that didn't happen in Python 2. Populating a new list of keys with `[*params]` solves this.